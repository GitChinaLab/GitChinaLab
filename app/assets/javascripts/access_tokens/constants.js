// Token types
export const FEED_TOKEN = 'feedToken';
export const INCOMING_EMAIL_TOKEN = 'incomingEmailToken';
export const STATIC_OBJECT_TOKEN = 'staticObjectToken';
