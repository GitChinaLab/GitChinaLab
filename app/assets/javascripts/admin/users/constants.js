import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};

export const CONFIRM_DELETE_BUTTON_SELECTOR = '.js-delete-user-modal-button';

export const MODAL_TEXTS_CONTAINER_SELECTOR = '#js-modal-texts';

export const MODAL_MANAGER_SELECTOR = '#js-delete-user-modal';
