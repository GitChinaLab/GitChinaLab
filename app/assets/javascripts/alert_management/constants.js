import { s__ } from '~/locale';

export const SEVERITY_LEVELS = {
  CRITICAL: s__('severity|Critical'),
  HIGH: s__('severity|High'),
  MEDIUM: s__('severity|Medium'),
  LOW: s__('severity|Low'),
  INFO: s__('severity|Info'),
  UNKNOWN: s__('severity|Unknown'),
};

export const ALERTS_STATUS_TABS = [
  {
    title: s__('AlertManagement|Open'),
    status: 'OPEN',
    filters: ['TRIGGERED', 'ACKNOWLEDGED'],
  },
  {
    title: s__('AlertManagement|Triggered'),
    status: 'TRIGGERED',
    filters: 'TRIGGERED',
  },
  {
    title: s__('AlertManagement|Acknowledged'),
    status: 'ACKNOWLEDGED',
    filters: 'ACKNOWLEDGED',
  },
  {
    title: s__('AlertManagement|Resolved'),
    status: 'RESOLVED',
    filters: 'RESOLVED',
  },
  {
    title: s__('AlertManagement|All alerts'),
    status: 'ALL',
    filters: ['TRIGGERED', 'ACKNOWLEDGED', 'RESOLVED'],
  },
];

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts list
 */
export const trackAlertListViewsOptions = {
  category: 'Alert Management',
  action: 'view_alerts_list',
};
