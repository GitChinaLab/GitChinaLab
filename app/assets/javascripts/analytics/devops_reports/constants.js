import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Reports');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Reports give an overview of how you are using GitLab from a feature perspective. Use them to view how you compare with other organizations, and how your teams compare against each other.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
