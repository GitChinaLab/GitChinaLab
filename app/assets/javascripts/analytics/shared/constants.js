import { masks } from 'dateformat';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};
