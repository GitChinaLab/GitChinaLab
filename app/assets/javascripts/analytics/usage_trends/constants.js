import { getDateInPast } from '~/lib/utils/datetime_utility';

export const TOTAL_DAYS_TO_SHOW = 365;
export const TODAY = new Date();
export const START_DATE = getDateInPast(TODAY, TOTAL_DAYS_TO_SHOW);
