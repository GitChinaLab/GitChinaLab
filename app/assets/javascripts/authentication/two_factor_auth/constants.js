export const COPY_BUTTON_ACTION = 'copy';
export const DOWNLOAD_BUTTON_ACTION = 'download';
export const PRINT_BUTTON_ACTION = 'print';

export const TRACKING_LABEL_PREFIX = '2fa_recovery_codes_';

export const RECOVERY_CODE_DOWNLOAD_FILENAME = 'gitlab-recovery-codes.txt';

export const SUCCESS_QUERY_PARAM = 'two_factor_auth_enabled_successfully';

export const COPY_KEYBOARD_SHORTCUT = 'mod+c';
