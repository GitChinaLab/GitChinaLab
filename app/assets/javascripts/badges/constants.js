export const GROUP_BADGE = 'group';
export const PROJECT_BADGE = 'project';
