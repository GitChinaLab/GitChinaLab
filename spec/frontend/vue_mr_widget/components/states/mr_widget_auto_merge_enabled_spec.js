import { shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';
import { trimText } from 'helpers/text_helper';
import { extendedWrapper } from 'helpers/vue_test_utils_helper';
import autoMergeEnabledComponent from '~/vue_merge_request_widget/components/states/mr_widget_auto_merge_enabled.vue';
import { MWPS_MERGE_STRATEGY } from '~/vue_merge_request_widget/constants';
import eventHub from '~/vue_merge_request_widget/event_hub';
import MRWidgetService from '~/vue_merge_request_widget/services/mr_widget_service';

let wrapper;
let mergeRequestWidgetGraphqlEnabled = false;

function convertPropsToGraphqlState(props) {
  return {
    autoMergeStrategy: props.autoMergeStrategy,
    cancelAutoMergePath: 'http://text.com',
    mergeUser: {
      id: props.mergeUserId,
      ...props.setToAutoMergeBy,
    },
    targetBranch: props.targetBranch,
    targetBranchCommitsPath: props.targetBranchPath,
    shouldRemoveSourceBranch: props.shouldRemoveSourceBranch,
    forceRemoveSourceBranch: props.shouldRemoveSourceBranch,
    userPermissions: {
      removeSourceBranch: props.canRemoveSourceBranch,
    },
  };
}

function factory(propsData, stateOverride = {}) {
  let state = {};

  if (mergeRequestWidgetGraphqlEnabled) {
    state = { ...convertPropsToGraphqlState(propsData), ...stateOverride };
  }

  wrapper = extendedWrapper(
    shallowMount(autoMergeEnabledComponent, {
      propsData: {
        mr: propsData,
        service: new MRWidgetService({}),
      },
      data() {
        return { state };
      },
      provide: { glFeatures: { mergeRequestWidgetGraphql: mergeRequestWidgetGraphqlEnabled } },
      mocks: {
        $apollo: {
          queries: {
            state: { loading: false },
          },
        },
      },
    }),
  );
}

const targetBranchPath = '/foo/bar';
const targetBranch = 'foo';
const sha = '1EA2EZ34';
const defaultMrProps = () => ({
  shouldRemoveSourceBranch: false,
  canRemoveSourceBranch: true,
  canCancelAutomaticMerge: true,
  mergeUserId: 1,
  currentUserId: 1,
  setToAutoMergeBy: {},
  sha,
  targetBranchPath,
  targetBranch,
  autoMergeStrategy: MWPS_MERGE_STRATEGY,
});

const getStatusText = () => wrapper.findByTestId('statusText').attributes('message');

describe('MRWidgetAutoMergeEnabled', () => {
  let oldWindowGl;

  beforeEach(() => {
    jest.spyOn(eventHub, '$emit').mockImplementation(() => {});

    oldWindowGl = window.gl;
    window.gl = {
      mrWidgetData: {
        defaultAvatarUrl: 'no_avatar.png',
      },
    };
  });

  afterEach(() => {
    window.gl = oldWindowGl;
    wrapper.destroy();
    wrapper = null;
  });

  [true, false].forEach((mergeRequestWidgetGraphql) => {
    describe(`when graphql is ${mergeRequestWidgetGraphql ? 'enabled' : 'disabled'}`, () => {
      beforeEach(() => {
        mergeRequestWidgetGraphqlEnabled = mergeRequestWidgetGraphql;
      });

      describe('computed', () => {
        describe('canRemoveSourceBranch', () => {
          it('should return true when user is able to remove source branch', () => {
            factory({
              ...defaultMrProps(),
            });

            expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(true);
          });

          it.each`
            mergeUserId | currentUserId
            ${2}        | ${1}
            ${1}        | ${2}
          `(
            'should return false when user id is not the same with who set the MWPS',
            ({ mergeUserId, currentUserId }) => {
              factory({
                ...defaultMrProps(),
                mergeUserId,
                currentUserId,
              });

              expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(false);
            },
          );

          it('should not find "Delete" button when shouldRemoveSourceBranch set to true', () => {
            factory({
              ...defaultMrProps(),
              shouldRemoveSourceBranch: true,
            });

            expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(false);
          });

          it('should find "Delete" button when shouldRemoveSourceBranch overrides state.forceRemoveSourceBranch', () => {
            factory(
              {
                ...defaultMrProps(),
                shouldRemoveSourceBranch: false,
              },
              {
                forceRemoveSourceBranch: true,
              },
            );

            expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(true);
          });

          it('should find "Delete" button when shouldRemoveSourceBranch set to false', () => {
            factory({
              ...defaultMrProps(),
              shouldRemoveSourceBranch: false,
            });

            expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(true);
          });

          it('should return false if user is not able to remove the source branch', () => {
            factory({
              ...defaultMrProps(),
              canRemoveSourceBranch: false,
            });

            expect(wrapper.findByTestId('removeSourceBranchButton').exists()).toBe(false);
          });
        });

        describe('cancelButtonText', () => {
          it('should return "Cancel" if MWPS is selected', () => {
            factory({
              ...defaultMrProps(),
              autoMergeStrategy: MWPS_MERGE_STRATEGY,
            });

            expect(wrapper.findByTestId('cancelAutomaticMergeButton').text()).toBe(
              'Cancel auto-merge',
            );
          });
        });
      });

      describe('methods', () => {
        describe('cancelAutomaticMerge', () => {
          it('should set flag and call service then tell main component to update the widget with data', (done) => {
            factory({
              ...defaultMrProps(),
            });
            const mrObj = {
              is_new_mr_data: true,
            };
            jest.spyOn(wrapper.vm.service, 'cancelAutomaticMerge').mockReturnValue(
              new Promise((resolve) => {
                resolve({
                  data: mrObj,
                });
              }),
            );

            wrapper.vm.cancelAutomaticMerge();
            setImmediate(() => {
              expect(wrapper.vm.isCancellingAutoMerge).toBeTruthy();
              if (mergeRequestWidgetGraphql) {
                expect(eventHub.$emit).toHaveBeenCalledWith('MRWidgetUpdateRequested');
              } else {
                expect(eventHub.$emit).toHaveBeenCalledWith('UpdateWidgetData', mrObj);
              }
              done();
            });
          });
        });

        describe('removeSourceBranch', () => {
          it('should set flag and call service then request main component to update the widget', (done) => {
            factory({
              ...defaultMrProps(),
            });
            jest.spyOn(wrapper.vm.service, 'merge').mockReturnValue(
              Promise.resolve({
                data: {
                  status: MWPS_MERGE_STRATEGY,
                },
              }),
            );

            wrapper.vm.removeSourceBranch();
            setImmediate(() => {
              expect(eventHub.$emit).toHaveBeenCalledWith('MRWidgetUpdateRequested');
              expect(wrapper.vm.service.merge).toHaveBeenCalledWith({
                sha,
                auto_merge_strategy: MWPS_MERGE_STRATEGY,
                should_remove_source_branch: true,
              });
              done();
            });
          });
        });
      });

      describe('template', () => {
        it('should have correct elements', () => {
          factory({
            ...defaultMrProps(),
          });

          expect(wrapper.element).toMatchSnapshot();
        });

        it('should disable cancel auto merge button when the action is in progress', async () => {
          factory({
            ...defaultMrProps(),
          });
          wrapper.setData({
            isCancellingAutoMerge: true,
          });

          await nextTick();

          expect(wrapper.find('.js-cancel-auto-merge').props('loading')).toBe(true);
        });

        it('should show source branch will be deleted text when it source branch set to remove', () => {
          factory({
            ...defaultMrProps(),
            shouldRemoveSourceBranch: true,
          });

          const normalizedText = wrapper.text().replace(/\s+/g, ' ');

          expect(normalizedText).toContain('Deletes the source branch');
          expect(normalizedText).not.toContain('Does not delete the source branch');
        });

        it('should not show delete source branch button when user not able to delete source branch', () => {
          factory({
            ...defaultMrProps(),
            currentUserId: 4,
          });

          expect(wrapper.find('.js-remove-source-branch').exists()).toBe(false);
        });

        it('should disable delete source branch button when the action is in progress', async () => {
          factory({
            ...defaultMrProps(),
          });
          wrapper.setData({
            isRemovingSourceBranch: true,
          });

          await nextTick();

          expect(wrapper.find('.js-remove-source-branch').props('loading')).toBe(true);
        });

        it('should render the status text as "...to merged automatically" if MWPS is selected', () => {
          factory({
            ...defaultMrProps(),
            autoMergeStrategy: MWPS_MERGE_STRATEGY,
          });

          expect(getStatusText()).toBe(
            'Set by %{merge_author} to be merged automatically when the pipeline succeeds',
          );
        });

        it('should render the cancel button as "Cancel" if MWPS is selected', () => {
          factory({
            ...defaultMrProps(),
            autoMergeStrategy: MWPS_MERGE_STRATEGY,
          });

          const cancelButtonText = trimText(wrapper.find('.js-cancel-auto-merge').text());

          expect(cancelButtonText).toBe('Cancel auto-merge');
        });
      });
    });
  });
});
