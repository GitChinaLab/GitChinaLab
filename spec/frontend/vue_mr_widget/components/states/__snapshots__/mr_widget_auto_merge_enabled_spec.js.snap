// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MRWidgetAutoMergeEnabled when graphql is disabled template should have correct elements 1`] = `
<div
  class="mr-widget-body media"
>
  <gl-icon-stub
    class="gl-text-blue-500 gl-mr-3 gl-mt-1"
    name="status_scheduled"
    size="24"
  />
   
  <div
    class="media-body"
  >
    <h4
      class="gl-display-flex"
    >
      <span
        class="gl-mr-3"
      >
        <gl-sprintf-stub
          data-testid="statusText"
          message="Set by %{merge_author} to be merged automatically when the pipeline succeeds"
        />
      </span>
       
      <gl-button-stub
        buttontextclasses=""
        category="primary"
        class="js-cancel-auto-merge"
        data-qa-selector="cancel_auto_merge_button"
        data-testid="cancelAutomaticMergeButton"
        icon=""
        size="small"
        variant="default"
      >
        
          Cancel auto-merge
        
      </gl-button-stub>
    </h4>
     
    <section
      class="mr-info-list"
    >
      <p
        class="gl-display-flex"
      >
        <span
          class="gl-mr-3"
        >
          Does not delete the source branch
        </span>
         
        <gl-button-stub
          buttontextclasses=""
          category="primary"
          class="js-remove-source-branch"
          data-testid="removeSourceBranchButton"
          icon=""
          size="small"
          variant="default"
        >
          
            Delete source branch
          
        </gl-button-stub>
      </p>
    </section>
  </div>
</div>
`;

exports[`MRWidgetAutoMergeEnabled when graphql is enabled template should have correct elements 1`] = `
<div
  class="mr-widget-body media"
>
  <gl-icon-stub
    class="gl-text-blue-500 gl-mr-3 gl-mt-1"
    name="status_scheduled"
    size="24"
  />
   
  <div
    class="media-body"
  >
    <h4
      class="gl-display-flex"
    >
      <span
        class="gl-mr-3"
      >
        <gl-sprintf-stub
          data-testid="statusText"
          message="Set by %{merge_author} to be merged automatically when the pipeline succeeds"
        />
      </span>
       
      <gl-button-stub
        buttontextclasses=""
        category="primary"
        class="js-cancel-auto-merge"
        data-qa-selector="cancel_auto_merge_button"
        data-testid="cancelAutomaticMergeButton"
        icon=""
        size="small"
        variant="default"
      >
        
          Cancel auto-merge
        
      </gl-button-stub>
    </h4>
     
    <section
      class="mr-info-list"
    >
      <p
        class="gl-display-flex"
      >
        <span
          class="gl-mr-3"
        >
          Does not delete the source branch
        </span>
         
        <gl-button-stub
          buttontextclasses=""
          category="primary"
          class="js-remove-source-branch"
          data-testid="removeSourceBranchButton"
          icon=""
          size="small"
          variant="default"
        >
          
            Delete source branch
          
        </gl-button-stub>
      </p>
    </section>
  </div>
</div>
`;
